### ISSUE TITLE
Possible error in `supportsFeature` function

### ISSUE THREAD

**Issue by jameskentTX** [pazznetwork/ngx-chat#31]
```typescript
    supportsFeature(jid: string, feature: string): Promise<boolean> {

        return new Promise((resolve, reject) => {

            this.servicesInitialized$.pipe(first(value => !!value)).subscribe(async () => {
                try {
                    const service = this.resourceCache[jid] || await this.discoverServiceInformation(jid);
                    if (!service) {
                        reject(new Error('no service found for jid ' + jid));
                    }
                    const results = this.resourceCache[jid].features.filter(resource => resource.indexOf(feature) >= 0);
                    resolve(results.length > 0);
                } catch (e) {
                    reject(e);
                }
            });

        });

    }
```

I compiled new version of ejabberd and I've made changes in configuration so I know now nothing about anything :) If I see any errors I don't know what is responsible for them.

But your function is throwing errors. I think this line is responsible:
```typescript
const results = this.resourceCache[jid].features.filter(resource => resource.indexOf(feature) >= 0);
```

It should be:
```typescript
const results = service.features.filter(resource => resource.indexOf(feature) >= 0);
```

**Comment by trampi** [comment]
Hi @jameskentTX!

While it is a code smell to not use the `service` variable, the code is actually exactly the same, the proposed change does not alter the behavior.

Which error do you get? The `no service found for jid`-error?

**Comment by jameskentTX** [comment]
Uncaught (in promise): TypeError: Cannot read property 'features' of undefined
TypeError: Cannot read property 'features' of undefined
    at ServiceDiscoveryPlugin.<anonymous> (service-discovery.plugin.ts:80)

**Comment by ganySA** [comment]
I have the exact same issue.

**Comment by trampi** [comment]
I've identified the root cause of the issue and will provide a fix in the next release.

**Comment by trampi** [comment]
@ganySA @jameskentTX 

Fix released in v0.9.2.
It is possible that your issue might still exist, should this be the case please reopen this ticket.
